services/retentionmgr/metadataDep.go

```diff
@@ -394,12 +394,21 @@ func (t *metadataDepImpl) GetAckLevel(destID destinationID, extID extentID, cgID
 	// assert(resp.GetExtent().GetExtentUUID() == extID)
 	// assert(resp.GetExtent().GetConsumerGroupUUID() == cgID)
 
-	// check if the consumer-group has read to "sealed" point
-	if resp.GetExtent().GetStatus() != metadata.ConsumerGroupExtentStatus_OPEN {
-		ackLevel = store.ADDR_SEAL
-	} else {
+	switch resp.GetExtent().GetStatus() {
+
+	case metadata.ConsumerGroupExtentStatus_OPEN:
 		ackLevel = resp.GetExtent().GetAckLevelOffset()
-		// assert(ackLevel != cherami.ADDR_END
+
+	case metadata.ConsumerGroupExtentStatus_CONSUMED:
```

Comment:
can we add some comments here regarding why the ack level is set to seal or begin?
also the change doesn't affect the behavior because you won't call this for deleted cg, right?